test/integration/puppetlabs/services/jruby_pool_manager/jruby_internal_test.clj

```diff
@@ -62,3 +61,19 @@
                                 profiler-files)]
         (is (= 2 (count profiler-files)))
         (is (not-empty (slurp real-profiler-file)))))))
+
+(deftest default-compile-mode
+  (testing "default compile-mode changes based on jruby version"
+    (let [pool (JRubyPool. 1)
+          config (logutils/with-test-logging
+                  (jruby-testutils/jruby-config {}))
+          instance (jruby-internal/create-pool-instance! pool 0 config #())
+          container (:scripting-container instance)]
+      (try
+        (if jruby-schemas/using-jruby-9k?
```

Comment:
I may not be understanding how this test works, but won't this only test one version of this conditional? I don't see how this test would get run twice.

Reply:
It doesn't get run twice, but there is a jruby9k profile that will also be run.

Reply:
(and travis runs both the standard 1.7 profile as well as the 9k profile https://github.com/puppetlabs/jruby-utils/blob/master/ext/travisci/test.sh#L8-L12)

Reply:
Ah I see, that travis info is helpful, thank you